sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -96,6 +97,7 @@ case class CreateViewCommand(
     qe.assertAnalyzed()
     val analyzedPlan = qe.analyzed
 
+    assertNoNullTypeInSchema(analyzedPlan.schema)
```

Comment:
This check is for `df.createTempView`. Note that, we support `select null as c` but fail in create view with null type plan.